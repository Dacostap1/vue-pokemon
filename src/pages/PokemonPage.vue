<template>
  <div v-if="!pokemon">
    <h1>Espere por favor..</h1>
  </div>
  <div v-else>
    <PokemonImage :pokemonId="this.pokemon.id" :showPokemon="showPokemon" />
    <PokemonOptions
      :pokemonList="pokemonList"
      @selectPokemon="pokemonSelected"
    />
    <template v-if="showAnswer">
      <h2 class="fade-in">{{ message }}</h2>
      <button @click="resetGame">Reset</button>
    </template>
  </div>
</template>

<script>
import PokemonImage from "@/components/PokemonImage.vue";
import PokemonOptions from "@/components/PokemonOptions.vue";
import { getPokemonOptions } from "@/helpers/getPokemonOptions";
export default {
  data() {
    return {
      pokemonList: [],
      pokemon: null,
      showPokemon: false,
      message: null,
      showAnswer: false,
    };
  },
  components: {
    PokemonImage,
    PokemonOptions,
  },
  mounted() {
    this.getPoke();
  },
  methods: {
    async getPoke() {
      this.pokemonList = await getPokemonOptions();
      const randomInt = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonList[randomInt];
    },
    pokemonSelected(pokeId) {
      //el $event tiene el valor del evento, puede ser (param1, param2, $event)
      this.showPokemon = true;
      this.showAnswer = true
      console.log(pokeId);

      if (pokeId == this.pokemon.id) {
        this.message = `Correcto, el pokemon era ${this.pokemon.name}`;
      } else {
        this.message = `Ops, el pokemon era ${this.pokemon.name}`;
      }
    },
    resetGame() {
      this.pokemonList = [];
      this.pokemon = null;
      this.showPokemon = false;
      this.message = null;
      this.showAnswer = false,

      this.getPoke()
    },
  },
};
</script>

<style></style>
