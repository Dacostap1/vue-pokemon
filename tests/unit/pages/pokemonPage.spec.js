import PokemonPage from "@/pages/PokemonPage";
import { mount, shallowMount } from "@vue/test-utils";
import { mockPokemons } from "../mocks/pokemons.mock";

describe("Test PokemonPage", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("debe de hacer match con el snapshot", () => {
    //No ayuda mucho porq solo se renderiza el v-if
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de llamar el metodo del mounted", () => {
    const getPoke = jest.spyOn(PokemonPage.methods, "getPoke");
    //Tiene que antes montarse el componente despues del spy
    const wrapper = shallowMount(PokemonPage);

    console.log(wrapper);
    expect(getPoke).toHaveBeenCalled();
  });

  test("debe de hacer match con el snapshot uego de cargar los pokemons", () => {
    //el shallowMount hace un mount suave sin entrar a los componentes internos, en lugar de eso les coloca una etiqueta stub

    const wrapper = mount(PokemonPage, {
      data() {
        return {
          pokemonList: mockPokemons,
          pokemon: mockPokemons[0],
          showPokemon: false,
          message: null,
          showAnswer: false,
        };
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
