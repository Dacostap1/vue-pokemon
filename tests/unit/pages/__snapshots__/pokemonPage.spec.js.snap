// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PokemonPage debe de hacer match con el snapshot 1`] = `
<div>
  <h1>Espere por favor..</h1>
</div>
`;

exports[`Test PokemonPage debe de hacer match con el snapshot uego de cargar los pokemons 1`] = `
<div>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/5.svg" alt="picture" class="hidden-pokemon">
    <!--v-if-->
  </div>
  <div class="options-container">
    <ul>
      <li>pikachu</li>
      <li>charmander</li>
      <li>venusaur</li>
      <li>new</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
